### ISSUE TITLE
Windows implementation

### ISSUE THREAD

**Issue by MOZGIII** [vasi/positioned-io#15]
It looks like windows implementation has a huge overhead. Let's discuss possible performance improvements!

It seems like it should be possible to use `WriteFileEx` and completion API. This also gives the benefit of having the ability to perform the operations asynchronously.

Additionally, it seems like a good idea to investigate the IOCP (and io_uring on Linux) as a completion based APIs.

**Comment by niklasf** [comment]
Overhead is much improved with the https://docs.rs/positioned-io-preview/0.3.4/positioned_io_preview/struct.RandomAccessFile.html wrapper from the preview branch. But `WriteFileEx` etc. are still preferable, probably.

**Comment by MOZGIII** [comment]
Yep, I think this crate should try to expose the positioned API, as mmap is already covered elsewhere:
https://github.com/RazrFalcon/memmap2-rs/blob/master/src/windows.rs
